File: CodingSamples/Java/Language-Runtime/AnnotationTest/demo2/services/CasualGreeter.java
package services;

import hosting.OnePerCall;

@OnePerCall
public class CasualGreeter implements Greeter {
    
    public CasualGreeter() {
        System.out.println("CasualGreeter activated.");
    }

    public String greet(String person) {
        return "Hi " + person;
    }
}

File: CodingSamples/DotNet/Language-Runtime/CustomAttributeTest/DemoApp2/Services/CasualGreeter.cs
using DemoApp.Hosting;

namespace DemoApp.Services;

[OnePerCall]
public class CasualGreeter : IGreeter, IDisposable
{
    public CasualGreeter()
    {
        Console.WriteLine("CasualGreeter activated");
    }

    public void Dispose()
    {
        Console.WriteLine("CasualGreeter deactivated");
    }

    public string Greet(string person)
    {
        return $"Hi {person}";
    }

    
}
